### PR TITLE
Integration of Systemtap as a profiling mechanism.

### PR DESCRIPTION
**Pull request by dgquintas**
Work remains to be done to:

1) Fix the makefile so that the .d -> .h compilation (via dtrace) happens.
2) Ability to narrow down which tags we are interested in for a given run. As things are now, all markings across the codebase are triggered, resulting in too high  overhead for marks we may not care about all the time.

### CODE REVIEW

File: src/core/profiling/timers.h [hunk 1]
```diff
@@ -34,35 +34,78 @@
...
-typedef struct grpc_timers_log grpc_timers_log;
+void grpc_timer_add_mark(int tag, void* id, const char *file, int line);
+void grpc_timer_begin(int tag, void* id, const char *file, int line);
+void grpc_timer_end(int tag, void* id, const char *file, int line);
 
-grpc_timers_log* grpc_timers_log_create(int capacity_limit, FILE* dump);
-void grpc_timers_log_add(grpc_timers_log*, const char* tag, void* id,
-                         const char* file, int line);
-void grpc_timers_log_destroy(grpc_timers_log *);
+enum profiling_tags {
```
Comment (ctiller) [inline src/core/profiling/timers.h]: grpc_profiling_tags

File: src/core/profiling/timers.h [hunk 2]
```diff
@@ -34,35 +34,78 @@
...
+  /* Re. Protos. */
+  PTAG_PROTO_SERIALIZE = 100,
```
Comment (ctiller) [inline src/core/profiling/timers.h]: GRPC_PTAG_xxxx

File: src/core/surface/channel_create.c [hunk 3]
```diff
@@ -195,6 +198,7 @@ grpc_channel *grpc_channel_create(const char *target,
 #define MAX_FILTERS 3
   const grpc_channel_filter *filters[MAX_FILTERS];
   int n = 0;
+  GRPC_TIMER_BEGIN(PTAG_OTHER_BASE + 2, NULL);
```
Comment (ctiller) [inline src/core/surface/channel_create.c]: Let's remove this before the commit: I think in any work we do for now, channel creation will be in the noise.

**Comment by ctiller** [comment]
Minor nits - ping me when the outstanding things are done and I'll get this merged.

**Comment by ctiller** [comment]
Can you merge with head and re push? Suspect Travis flakes will somewhat disappear with Nicolas latest work.

**Comment by nicolasnoble** [comment]
These don't really seem like flakes to me:

```
test/core/profiling/timers_test.c: In function ‘main’:
test/core/profiling/timers_test.c:79:3: error: implicit declaration of function ‘grpc_timers_log_global_init’ [-Werror=implicit-function-declaration]
test/core/profiling/timers_test.c:81:3: error: implicit declaration of function ‘grpc_timers_log_global_destroy’ [-Werror=implicit-function-declaration]
```

:)

**Comment by ctiller** [comment]
I missed that one amongst other noise.

**Comment by dgquintas** [comment]
Woops, fixed. I forgot to build the tests. I've re-merged and pushed just
now.

**Pull request merged**